readthedocs/proxito/middleware.py

```diff
@@ -218,9 +218,16 @@ def process_request(self, request):  # noqa
                 context={"host": exc.domain},
                 status=400,
             )
-        except (InvalidSubdomainError, InvalidExternalDomainError):
+        except (InvalidSubdomainError, InvalidExternalDomainError) as exc:
             log.debug("Invalid project set on the subdomain.")
-            raise Http404
+            # We pass e.domain as 'project_slug' because we cannot unpack a slug from the
+            # domain at this stage. The domain can be a custom domain docs.foobar.tld,
+            # in which case 'docs' would be an incorrect slug.
+            raise ProxitoProjectHttp404(
```

Comment:
I guess because we try to limit the scope, I'm not gonna look at it this time around, but will be easy to add if anyone bumps into a good use case :+1: